File: ungs-docsys-bff/src/modules/resume-user/dtos/resume-user-response.dto.ts
import { ContactResponseDto } from './contact-response.dto';
import { EducationResponseDto } from './education-response.dto';
import { ExperienceResponseDto } from './experience-response.dto';
import { LanguageResponseDto } from './language-response.dto';
import { TechnicalSkillResponseDto } from './technical-skill-response.dto';
import { CertificationResponseDto } from './certification-response.dto';
import { ResumeFileResponseDto } from './resume-file-response.dto';
import { AppUserExtendedResponseDto } from '../../../modules/login/app-user/dtos/app-user-extended-response.dto';
import { UserInfoResponseDto } from '../../../modules/login/user-info/dtos/user-info-response.dto';

export class ResumeUserResponseDto {
  id: number;
  contact: ContactResponseDto;
  educations: EducationResponseDto[];
  experiences: ExperienceResponseDto[];
  languages: LanguageResponseDto[];
  technicalSkills: TechnicalSkillResponseDto[];
  certifications: CertificationResponseDto[];
  resumeFiles: ResumeFileResponseDto[];
  appUser: AppUserExtendedResponseDto;
  userInfo: UserInfoResponseDto;
}

File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/ResumeUserResponseDto.java
package com.ungs.docsys.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResumeUserResponseDto {
    private Long id;
    private ContactResponseDto contact;
    private List<EducationResponseDto> educations;
    private List<ExperienceResponseDto> experiences;
    private List<LanguageResponseDto> languages;
    private List<TechnicalSkillResponseDto> technicalSkills;
    private List<CertificationResponseDto> certifications;
    private List<ResumeFileResponseDto> resumeFiles;
    private AppUserResponseDto appUser;
    private UserInfoResponseDto userInfo;
}

